gadget/update_test.go

```diff
@@ -1186,6 +1237,35 @@ func (u *updateTestSuite) TestUpdateApplyUC20MissingInitialMapFullLogicOnlySyste
 	c.Assert(muo.canceledCalled, Equals, 0)
 }
 
+func mockOnDiskForMultiVolumeUC20GadgetYamlNoBIOS() map[string]map[int]*gadget.OnDiskStructure {
+	volToIdxToDiskStruct := map[string]map[int]*gadget.OnDiskStructure{
+		"pc": {
+			0: {
+				Name: "mbr",
+			},
+			1: {
+				Name: "ubuntu-save",
+			},
+			2: {
+				Name: "ubuntu-boot",
+			},
+			3: {
+				Name: "ubuntu-save",
+			},
+			4: {
+				Name: "ubuntu-data",
+			},
+		},
+		"foo": {
+			0: {Name: "barething"},
+			1: {Name: "nofspart"},
+			2: {Name: "some-filesystem"},
+		},
+	}
+
+	return volToIdxToDiskStruct
+}
```

Comment:
couldn't gadgettest.LayoutMultiVolumeFromYaml do this mocking by itself in many cases?